executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -284,6 +308,16 @@ pub trait FixedEvaluator<T: FieldElement> {
     }
 }
 
+pub struct FixedEvaluatorForFixedData<'a, T: FieldElement>(pub &'a FixedData<'a, T>);
```

Comment:
Hmm, I'm not sure I fully agree!

I understood the interface more like `evaluate` receives a *global* (but possibly negative index). It's the responsibility of the user of `WitgenInference` to pass a row that makes sense, and the block machine processor in #2226 would make the assumption that fixed columns are cyclic. That would make sense, no? Maybe then I should rename the argument from "row offset" to just "row"?

(2) is fixed.

Also, I could undo the change and discuss this in the next PR, because in this PR it's still only used in the test.